handlers/line/line.go

```diff
@@ -118,6 +129,50 @@ func (h *handler) receiveMessage(ctx context.Context, channel courier.Channel, w
 
 }
 
+func (h *handler) validateSignature(channel courier.Channel, r *http.Request) error {
+	actual := r.Header.Get(signatureHeader)
+	if actual == "" {
+		return fmt.Errorf("missing request signature")
+	}
+
+	confAuth := channel.ConfigForKey(courier.ConfigAuthToken, "")
+	authToken, isStr := confAuth.(string)
+	if !isStr || authToken == "" {
+		return fmt.Errorf("invalid or missing auth token in config")
+	}
+
+	expected, err := lineCalculateSignature(authToken, r)
+	if err != nil {
+		return err
+	}
+
+	// compare signatures in way that isn't sensitive to a timing attack
+	if !hmac.Equal(expected, []byte(actual)) {
+		return fmt.Errorf("invalid request signature")
+	}
+
+	return nil
+}
+
+// see https://developers.line.me/en/docs/messaging-api/reference/#signature-validation
+func lineCalculateSignature(authToken string, r *http.Request) ([]byte, error) {
```

Comment:
No need to stutter (that's a golang term), you are already in the Line package so no need to start with `line..`